src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/AutoCrafterListener.java

```diff
@@ -35,6 +43,15 @@ public AutoCrafterListener(SlimefunPlugin plugin) {
         plugin.getServer().getPluginManager().registerEvents(this, plugin);
     }
 
+    public boolean clickEventRecipeUnlockedCheck(PlayerRightClickEvent e) {
```

Comment:
Never pass an event through a method. Only pass the needed params. Also, we need a better name for this. A simple, `hasUnlockedRecipe` should be fine

Reply:
Agreed, I think `hasUnlockedRecipe(@Nonnull Player p, @Nonnull ItemStack item)` is sufficient.
Also that method should rather be `private` or `protected`

Reply:
Okay! Thanks, I will change it. Sorry I just wasn't really familiar with the methods stuff in like a big project like this.